src/common/api/fetch.ts

```diff
@@ -1,12 +1,16 @@
 import { HIRO_HEADERS, withApiServer } from '@/common/constants';
 
-export const fetcher = (input: RequestInfo, init: RequestInit = {}) => {
+export const fetcher = async (input: RequestInfo, init: RequestInit = {}) => {
   const initHeaders = init.headers || {};
-  return fetch(input, {
-    credentials: 'omit',
-    ...init,
-    headers: { ...initHeaders, ...HIRO_HEADERS },
-  });
+  try {
+    return await fetch(input, {
+      credentials: 'omit',
+      ...init,
+      headers: { ...initHeaders, ...HIRO_HEADERS },
+    });
+  } catch (error) {
+    throw new Error(`Network error: ${error.message}`);
```

Comment:
Hi @shamim-io! Thank you for submitting this PR. Could you explain the motivation behind this change? It looks like the error is being rethrown.

Reply:
@He1DAr If you think that, its not a required change. Feel free to close it :)